src/dashboard/src/components/api/views.py

```diff
@@ -149,31 +174,22 @@ def status(request, unit_uuid, unit_type):
     if error is not None:
         response['message'] = error
         response['error'] = True
-        return django.http.HttpResponseServerError(
+        return django.http.HttpResponseServerError(  # 500
             json.dumps(response),
             content_type='application/json'
         )
     response['message'] = 'Fetched status for {} successfully.'.format(unit_uuid)
     return helpers.json_response(response)
 
 
+@_api_endpoint(expected_methods=['GET'])
 def waiting_for_user_input(request):
-    # Example: http://127.0.0.1/api/transfer/waiting?username=mike&api_key=<API key>
-    if request.method not in ('GET',):
-        return django.http.HttpResponseNotAllowed(['GET'])
-
-    auth_error = authenticate_request(request)
+    # Example: http://127.0.0.1/api/ingest/waiting?username=mike&api_key=<API key>
     response = {}
-    if auth_error is not None:
-        response = {'message': auth_error, 'error': True}
-        return django.http.HttpResponseForbidden(
-            json.dumps(response),
-            content_type='application/json'
-        )
-
     error = None
```

Comment:
This is unrelated to your specific changes, but I'm not seeing how ``error`` could ever be modified between lines 174-192. Therefore, isn't the entire `if error is not None` block at 193 unnecessary?